discovery/discovery_test.go

```diff
@@ -350,20 +350,106 @@ func TestNewDBOrchestorPoolCache_PollOrchestratorInfo(t *testing.T) {
 		require.Contains(testOrchs, test)
 	}
 
-	cacheRefreshInterval = 1 * time.Second
+	cacheRefreshInterval = 1 * time.Millisecond
 	returnInfo = polledOrchInfo
-	pool.PollOrchestratorInfo(nil)
-	time.Sleep(3 * time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	pool.PollOrchestratorInfo(ctx, nil)
+	time.Sleep(5 * time.Millisecond)
 	dbOrchs, err = pool.store.SelectOrchs(nil)
 	require.Nil(err)
 	require.Len(dbOrchs, 3)
 	expPrice, _ = common.PriceToFixed(big.NewRat(1, 1))
 	for _, o := range dbOrchs {
-		t.Log(o)
 		assert.Equal(o.PricePerPixel, expPrice)
 	}
 }
 
+func TestNewDBOrchestorPoolCache_PollOrchestratorInfo_CallCount(t *testing.T) {
+	var mu sync.Mutex
+	first := true
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
+		mu.Lock()
+		if first {
+			time.Sleep(100 * time.Millisecond)
+			first = false
+		}
+		mu.Unlock()
+		return &net.OrchestratorInfo{
+			Transcoder: "transcoderFromTest",
+			PriceInfo: &net.PriceInfo{
+				PricePerUnit:  1,
+				PixelsPerUnit: 1,
+			},
+		}, nil
+	}
+
+	dbh, dbraw, err := common.TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	require := require.New(t)
+	assert := assert.New(t)
+	require.Nil(err)
+
+	// adding orchestrators to DB
+	addresses := []string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"}
+	orchestrators := StubOrchestrators(addresses)
+	testOrchs := make([]orchTest, 0)
+	for _, o := range orchestrators {
+		to := orchTest{
+			EthereumAddr:  o.Address.String(),
+			ServiceURI:    o.ServiceURI,
+			PricePerPixel: 1000,
+		}
+		testOrchs = append(testOrchs, to)
+	}
+
+	ethClient := &eth.StubClient{Orchestrators: orchestrators}
+
+	pool := NewDBOrchestratorPoolCache(dbh, ethClient, &stubRoundsManager{}, &stubTicketParamsValidator{})
+
+	// Add orchestrators to DB
+	err = pool.CacheTranscoderPool(nil)
+	require.Nil(err)
+
+	// Ensure orchestrators exist in DB
+	require.Equal(pool.Size(), 3)
+	dbOrchs, err := pool.store.SelectOrchs(nil)
+	require.Nil(err)
+	require.Len(dbOrchs, 3)
+	for _, o := range dbOrchs {
+		test := toOrchTest(o.EthereumAddr, o.ServiceURI, o.PricePerPixel)
+		require.Contains(testOrchs, test)
+	}
+
+	callCount := 0
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
```

Comment:
Instead of re-assigning this variable to a tweaked version of the original function defined further up in this test what if we include `callCount` in the original definition and then after the setup phase of this test (i.e. everything before this line) we store the current value of `callCount` and then after we run `PollOrchestratorInfo()` and wait through the sleep time, we compare it against the new value of `callCount`. I think this could help slim down this test a bit.

Reply:
Ah yes we could just reset `callCount` to 0 as well after calling `CacheTranscoderPool` as well in fact.